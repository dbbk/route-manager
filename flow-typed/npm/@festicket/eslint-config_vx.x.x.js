// flow-typed signature: 564086c0274f153b4c37ac5c954f219e
// flow-typed version: <<STUB>>/@festicket/eslint-config_v^1.1.5/flow_v0.63.1

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@festicket/eslint-config'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@festicket/eslint-config' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */


// Filename aliases
declare module '@festicket/eslint-config/index' {
  declare module.exports: $Exports<'@festicket/eslint-config'>;
}
declare module '@festicket/eslint-config/index.js' {
  declare module.exports: $Exports<'@festicket/eslint-config'>;
}
