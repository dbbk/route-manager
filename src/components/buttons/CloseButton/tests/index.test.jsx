// @flow

import React from 'react';
import { mount } from 'enzyme';
import { spy } from 'sinon';
import { CloseButton } from '@festicket/react-ui-components';

describe('<CloseButton />', () => {
  test('should check component snapshot', () => {
    const handle = spy();
    const component = mount(<CloseButton alt="Close" handleClose={handle} />);

    expect(component).toMatchSnapshot();
  });

  test('should check component snapshot with size', () => {
    const handle = spy();
    const component = mount(
      <CloseButton alt="Close" size={50} handleClose={handle} />,
    );

    expect(component).toMatchSnapshot();
  });

  test('should call handleClose on click', () => {
    const handle = spy();
    const wrapper = mount(<CloseButton alt="Close" handleClose={handle} />);
    expect(handle.callCount).toEqual(0);
    wrapper.simulate('click');
    expect(handle.callCount).toEqual(1);
  });
});
