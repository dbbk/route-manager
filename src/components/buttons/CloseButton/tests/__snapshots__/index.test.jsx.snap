// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseButton /> should check component snapshot 1`] = `
.c1 {
  fill: currentColor;
}

.c0 {
  font-size: 18px;
  background-color: transparent;
  border: 0;
  cursor: pointer;
  padding: 0;
  outline: 0;
  overflow: hidden;
}

<CloseButton
  alt="Close"
  handleClose={[Function]}
>
  <Styled(CloseButtonComponent)
    alt="Close"
    handleClose={[Function]}
    size={18}
  >
    <CloseButtonComponent
      alt="Close"
      className="c0"
      handleClose={[Function]}
      size={18}
    >
      <button
        aria-label="Close"
        className="c0"
        onClick={[Function]}
      >
        <Styled(Close)>
          <Close
            className="c1"
          >
            <svg
              className="c1"
              height="1em"
              viewBox="0 0 32 32"
              width="1em"
            >
              <path
                d="M27.893 25.262L18.65 16l9.26-9.262a1.874 1.874 0 0 0-2.647-2.647L16 13.353 6.74 4.108a1.867 1.867 0 0 0-2.63 2.636l9.242 9.26-9.244 9.262c-.34.34-.55.807-.55 1.324a1.874 1.874 0 0 0 3.198 1.324L16 18.65l9.262 9.262c.34.34.807.55 1.324.55a1.874 1.874 0 0 0 1.323-3.198z"
              />
            </svg>
          </Close>
        </Styled(Close)>
      </button>
    </CloseButtonComponent>
  </Styled(CloseButtonComponent)>
</CloseButton>
`;

exports[`<CloseButton /> should check component snapshot with size 1`] = `
.c1 {
  fill: currentColor;
}

.c0 {
  font-size: 50px;
  background-color: transparent;
  border: 0;
  cursor: pointer;
  padding: 0;
  outline: 0;
  overflow: hidden;
}

<CloseButton
  alt="Close"
  handleClose={[Function]}
  size={50}
>
  <Styled(CloseButtonComponent)
    alt="Close"
    handleClose={[Function]}
    size={50}
  >
    <CloseButtonComponent
      alt="Close"
      className="c0"
      handleClose={[Function]}
      size={50}
    >
      <button
        aria-label="Close"
        className="c0"
        onClick={[Function]}
      >
        <Styled(Close)>
          <Close
            className="c1"
          >
            <svg
              className="c1"
              height="1em"
              viewBox="0 0 32 32"
              width="1em"
            >
              <path
                d="M27.893 25.262L18.65 16l9.26-9.262a1.874 1.874 0 0 0-2.647-2.647L16 13.353 6.74 4.108a1.867 1.867 0 0 0-2.63 2.636l9.242 9.26-9.244 9.262c-.34.34-.55.807-.55 1.324a1.874 1.874 0 0 0 3.198 1.324L16 18.65l9.262 9.262c.34.34.807.55 1.324.55a1.874 1.874 0 0 0 1.323-3.198z"
              />
            </svg>
          </Close>
        </Styled(Close)>
      </button>
    </CloseButtonComponent>
  </Styled(CloseButtonComponent)>
</CloseButton>
`;
